jsonalchemy/wrappers.py

```diff
@@ -60,6 +60,14 @@ def __init__(self, schema=None, root=None):
         else:
             self.root = self
 
+    def search(self, query):
+        jsonpath_expr = parse(query)
+        result = [match.value for match in jsonpath_expr.find(self)]
+
+        return wrap(result,
+                    {'type': 'array', 'items': [el.schema for el in result]},
+                    result)
```

Comment:
Is this correct? `result` might not be the actual `root` the items in `JSONArray` are looking for when accessing calculated or derived fields.